lib/api-tests/tests/high_level_api.rs

```diff
@@ -0,0 +1,177 @@
+                test_table_seen = true;
+            }
+            ("test-global", export::ExportKind::Global) => {
+                assert!(!test_global_seen);
+                test_global_seen = true;
+            }
+            ("ret_2", export::ExportKind::Function) => {
+                assert!(!test_ret_2_seen);
+                test_ret_2_seen = true;
+            }
+            ("ret_4", export::ExportKind::Function) => {
+                assert!(!test_ret_4_seen);
+                test_ret_4_seen = true;
+            }
+            ("set_test_global", export::ExportKind::Function) => {
+                assert!(!set_test_global_seen);
+                set_test_global_seen = true;
+            }
+            ("update_outside_global", export::ExportKind::Function) => {
+                assert!(!update_outside_global_seen);
+                update_outside_global_seen = true;
+            }
+            _ => {
+                assert!(false, "Unknown export found!");
+            }
+        }
+    }
+    assert!(test_table_seen);
+    assert!(test_global_seen);
+    assert!(test_ret_2_seen);
+    assert!(test_ret_4_seen);
+    assert!(set_test_global_seen);
+    assert!(update_outside_global_seen);
+
+    assert_eq!(
+        module.custom_sections("test_custom_section"),
+        Some(&[b"hello, world!".to_vec(), b"goodbye, world!".to_vec()][..])
+    );
+    let outside_global = Global::new_mutable(Value::I32(15));
+
+    let import_object = imports! {
+        "env" => {
+            "update-func" => func!(|x: i32| x * 2),
+            "outside-global" => outside_global.clone(),
+        }
+    };
+    let instance = module.instantiate(&import_object).unwrap();
+
+    let ret_2: Func<(), i32> = instance.exports_new().get("ret_2").unwrap();
+    let ret_4: Func<(), i32> = instance.exports_new().get("ret_4").unwrap();
+    let set_test_global: Func<i32> = instance.exports_new().get("set_test_global").unwrap();
+    let update_outside_global: Func = instance.exports_new().get("update_outside_global").unwrap();
+
+    assert_eq!(ret_2.call(), Ok(2));
+    assert_eq!(ret_4.call(), Ok(4));
+
+    let _test_table: Table = instance.exports_new().get("test-table").unwrap();
+    // TODO: when table get is stablized test this
+
+    let test_global: Global = instance.exports_new().get("test-global").unwrap();
+    // TODO: do we want to make global.get act like exports_new().get()?
+    assert_eq!(test_global.get(), Value::I32(3));
+    set_test_global.call(17).unwrap();
+    assert_eq!(test_global.get(), Value::I32(17));
+
+    assert_eq!(outside_global.get(), Value::I32(15));
+    update_outside_global.call().unwrap();
+    assert_eq!(outside_global.get(), Value::I32(15 * 2));
+    update_outside_global.call().unwrap();
+    assert_eq!(outside_global.get(), Value::I32(15 * 2 * 2));
+}
+
+// copied from Rust stdlib: https://doc.rust-lang.org/nightly/nightly-rustc/src/serialize/leb128.rs.html#4-14
+macro_rules! impl_write_unsigned_leb128 {
```

Comment:
Why this is needed?

Reply:
It's not strictly needed, but in case we needed the other functions, I figured it might be good to have around.  I can macro-expand/inline the one function I need for the custom section writing if you'd prefer that!